src/category_theory/abelian/non_preadditive.lean

```diff
@@ -0,0 +1,720 @@
+/-
+Copyright (c) 2020 Markus Himmel. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Markus Himmel
+-/
+import category_theory.limits.shapes.binary_products
+import category_theory.limits.shapes.kernels
+import category_theory.limits.shapes.pullbacks
+import category_theory.limits.shapes.regular_mono
+import category_theory.preadditive
+
+/-!
+# Every non_preadditive_abelian category is preadditive
+
+In mathlib, we define an abelian category as a preadditive category with a zero object,
+kernels and cokernels, products and coproducts and in which every monomorphism and epimorphis is
+normal.
+
+While virtually every interesting abelian category has a natural preadditive structure (which is why
+it is included in the definition), preadditivity is not actually needed: Every category that has
+all of the other properties appearing in the definition of an abelian category admits a preadditive
+structure. This is the construction we carry out in this file.
+
+The proof proceeds in roughly five steps:
+1. Prove some results (for example that all equalizers exist) that would be trivial if we already
+   had the preadditive structure but are a bit of work without it.
+2. Copy-and-paste some results from the formalization of abelian categories in the main mathlib
+   repository.
```

Comment:
This no longer makes much sense, as you've ended up doing `cut-and-paste` instead.